src/utils/hooks/useGetResponsesByCode.tsx

```diff
@@ -0,0 +1,60 @@
+import { useState, useEffect } from 'react'
+import {
+  Application,
+  ApplicationResponse,
+  ApplicationSection,
+  useGetApplicationQuery,
+  GetApplicationQuery,
+} from '../generated/graphql'
+import { ResponsesByCode } from '../types'
+
+interface useLoadApplicationProps {
+  serialNumber: string
+}
+
+const useGetResponsesByCode = (props: useLoadApplicationProps) => {
+  const { serialNumber } = props
+  const [responsesByCode, setResponsesByCode] = useState({})
+  const [error, setError] = useState('')
+  const { data, loading: apolloLoading, error: apolloError } = useGetApplicationQuery({
+    variables: { serial: serialNumber },
+  })
+
+  useEffect(() => {
+    const error = checkForApplicationErrors(data)
+
+    if (error) {
+      setError(error)
+      return
+    }
+
+    const applicationResponses = data?.applications?.nodes[0]?.applicationResponses
+      .nodes as ApplicationResponse[]
+
+    const currentResponses = {} as ResponsesByCode
+
+    applicationResponses.forEach((response) => {
+      const code = response?.templateElement?.code
+      if (code) currentResponses[code] = response?.value?.text || response?.value
+    })
+
+    setResponsesByCode(currentResponses)
+  }, [data, apolloError])
+
+  return {
+    apolloError,
+    apolloLoading,
```

Comment:
It would be nice to also have the loading done here. Similar to what was done in `useLoadTemplate`, because the `apolloLoading` will be set to false as soon as the query is fetch, but we would like to watch a `loading` that is only set to be false after this hook has finished doing the manipulation of data.